File: languages/csharp/01-foundations/strings/example/main.cs
// Module focus: Cleaning and combining text while preserving readable string logic.
// Why it matters: the example makes it possible to normalize, inspect, and transform textual data
// before the learner tackles the exercises.

using System;
using System.Text;

// Separate helpers keep the main path focused on how to normalize, inspect, and transform textual
// data.
class Program
{
    // Fixed inputs make the consequence of counting words without removing extra spaces visible
    // and repeatable.
    static void Main()
    {
        // These values exercise the normal path before the exercises vary the documented
        // boundaries.
        // The printed result shows whether the program can handle empty input and character
        // boundaries safely.
        Console.Write("Enter a sentence: ");
        string line = Console.ReadLine() ?? "";

        StringBuilder cleanedBuilder = new StringBuilder();
        foreach (char ch in line)
        {
            if (char.IsLetterOrDigit(ch))
            {
                cleanedBuilder.Append(char.ToLowerInvariant(ch));
            }
            else
            {
                cleanedBuilder.Append(' ');
            }
        }

        string cleaned = cleanedBuilder.ToString();
        string[] words = cleaned.Split(' ', StringSplitOptions.RemoveEmptyEntries);

        Console.WriteLine($"Normalized text: {cleaned}");
        Console.WriteLine($"Tokens ({words.Length}):");
        foreach (string word in words)
        {
            Console.WriteLine($"- {word}");
        }
    }
}

File: languages/python/01-foundations/strings/example/main.py
# Module focus: Cleaning and combining text while preserving readable string logic.
# Why it matters: the example makes it possible to normalize, inspect, and transform textual data
# before the learner tackles the exercises.
# about.

# Fixed inputs make the consequence of counting words without removing extra spaces visible and
# repeatable.
line = input("Enter a sentence: ")

cleaned_chars = []
for char in line:
    if char.isalnum():
        cleaned_chars.append(char.lower())
    else:
        cleaned_chars.append(" ")

cleaned = "".join(cleaned_chars)
words = [word for word in cleaned.split() if word]

# The printed result shows whether the program can handle empty input and character boundaries
# safely.
print(f"Normalized text: {cleaned}")
print(f"Tokens ({len(words)}):")
for word in words:
    print(f"- {word}")
